### PR TITLE
W-13223477: ignore resources without actions

### PR DESCRIPTION
**Pull request by juanchib**

### CODE REVIEW

File: src/main/java/org/mule/module/apikit/FlowFinder.java [hunk 1]
```diff
@@ -99,7 +99,9 @@ private List<Flow> getFlows() {
 
   private void flattenResourceTree(Map<String, Resource> resources, String version) {
     for (Resource resource : resources.values()) {
-      flatResourceTree.put(resource.getResolvedUri(version), resource);
+      if (!resource.getActions().isEmpty()) {
+        flatResourceTree.put(resource.getResolvedUri(version), resource);
+      }
```
Comment (juanchib) [inline src/main/java/org/mule/module/apikit/FlowFinder.java:104]: this is the fix

File: src/test/java/org/mule/module/apikit/validation/AbstractRequestValidatorTestCase.java [hunk 2]
```diff
@@ -151,22 +151,27 @@ public TestRestRequestValidatorBuilder withCharset(String payloadCharset) {
...
-          .rawRequestPath(isNoneBlank(rawRequestPath) ? rawRequestPath : relativePath)
+          .rawRequestPath(rawRequestPath)
           .build();
       return new TestRestRequestValidator(relativePath, parser, create(apiLocation), charset, body, httpRequestAttributes,
                                           validationConfig);
     }
 
     private void validateMandatory() {
       assertNotNull("API location is mandatory for the test to run", apiLocation);
-      assertNotNull("Request path is mandatory for the test to run", requestPath);
```
[the remark this answers is not in the document]
Comment (dcourtil) [reply src/test/java/org/mule/module/apikit/validation/AbstractRequestValidatorTestCase.java:166]: What if we change this at the scaffolder at some point? Don't you think it gets too coupled to the scaffolder?

[the remark this answers is not in the document]
Comment (juanchib) [reply src/test/java/org/mule/module/apikit/validation/AbstractRequestValidatorTestCase.java:166]: The test?

File: src/test/java/org/mule/module/apikit/validation/MethodNotAllowedTestCase.java [hunk 3]
```diff
@@ -12,18 +12,35 @@
...
+        .build()
+        .validateRequest();
+
+    testRestRequestValidatorBuilder
+        .withApiLocation("unit/validation/api-resources.raml")
+        .withRelativePath("/test/something/else")
+        .withMethod("PUT")
+        .build()
+        .validateRequest();
+  }
```
Comment (juanchib) [inline src/test/java/org/mule/module/apikit/validation/MethodNotAllowedTestCase.java:45]: this are the tests I've added

File: src/test/java/org/mule/module/apikit/validation/AbstractRequestValidatorTestCase.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dcourtil) [reply src/test/java/org/mule/module/apikit/validation/AbstractRequestValidatorTestCase.java:166]: Dismiss my comment, it makes sense as we are hardcoding the path in the listener

**Review by dcourtil — APPROVED** [review]
LGTM

**Pull request merged**